Assets/Scripts/MineManager.cs

```diff
@@ -0,0 +1,267 @@
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.SceneManagement;
+using Random = UnityEngine.Random;
+
+
+public class MineManager : MonoBehaviour
+{
+    // model
+    public enum State 
+    {
+        Init,
+        Flagged,
+        Opened
+    }
+    public enum GameState 
+    {
+        Playing,
+        Win,
+        Lose
+    }
+    
+    
+    public int  gridSize;
+    public int mineCount;
+    public int[,] scoreGrid;
+    public State[,] gridStates;
+    public int opened = 0;
+    
+    private float time = 0;
+    private float delayTime = 1f;
+    
+    public GameState gameState = GameState.Playing;
+    
+    
+    //view
+    [SerializeField] private GameObject[] boxs;
+    [SerializeField] private Sprite flag;
+    [SerializeField] private Sprite[] number;
+    public Sprite defaultSprite;
+    public Color[] colors;
+    
+    public List<(Vector2Int, State)> Changelist = new List<(Vector2Int, State)>();
+    
+    
+    
+    public static MineManager Instance { get; private set; }
+
+    void Awake()
+    {
+        if (Instance == null)
+        {
+            Instance = this;
+            DontDestroyOnLoad(gameObject);
```

Comment:
i think this is the reason you had to destroy the object in EndController. what's the purpose of this?

Reply:
Otherwise, it will not be able to check the winning/losing state that is stored in the MineManager.
<img width="506" alt="image" src="https://github.com/Ad00re/GameProjects/assets/37167982/f759067a-cdde-46bb-a85c-0ca969d02354">